runtime/vm/jvminit.c

```diff
@@ -6000,6 +6000,270 @@ processCompressionOptions(J9JavaVM *vm){
 	}
 }
 
+/**
+ * For compatibility when -XX:-LegacyXlogOption is not set, parse -Xlog:gc options and add mappings.
+ * Anything besides -Xlog:gc, -Xlog:gc:stderr|<file>|file=<file>[:[:]] gives an unrecognized warning,
+ * although any superset such as -Xlog, -Xlog:all, -Xlog:something,gc, etc. enables gc logging as well. 
+ */
+static IDATA
+parseXlogForCompatibility(J9JavaVM *vm)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	IDATA rc = JNI_OK;
+	J9VMInitArgs *j9vm_args = vm->vmArgsArray;
+	IDATA xlogindex = FIND_ARG_IN_VMARGS_FORWARD(OPTIONAL_LIST_MATCH, MAPOPT_XLOG_OPT, NULL);
+	while (xlogindex >= 0) {
+		BOOLEAN unrecognizedOption = FALSE;
+		BOOLEAN setgclog = FALSE;
+		BOOLEAN gclog = FALSE;
+		char *gclogfile = NULL;
+		char xlogoptionsbuf[LARGE_STRING_BUF_SIZE];
+		char *xlogoptions = xlogoptionsbuf;
+		/* Include room for the trailing NULL */
+		UDATA optionLen = strlen(j9vm_args->actualVMArgs->options[xlogindex].optionString) + 1;
+
+		if (optionLen > sizeof(xlogoptionsbuf)) {
+			xlogoptions = j9mem_allocate_memory(optionLen, OMRMEM_CATEGORY_VM);
+			if (NULL == xlogoptions) {
+				goto xlogret;
+			}
+		}
+
+		if (OPTION_OK != COPY_OPTION_VALUE(xlogindex, ':', &xlogoptions, optionLen)) {
+			rc = JNI_ERR;
+			goto xlogret;
+		}
+
+		if (NULL == xlogoptions) {
+			/* Reached if just -Xlog is specified */
+			setgclog = TRUE;
+			gclog = TRUE;
+			unrecognizedOption = TRUE;
+		} else {
+			/* -Xlog:gc+other=off,a+b=c:file=abc:dec:options */
+			char *fullOptions = xlogoptions;
+			char *upToColon = strsep(&fullOptions, ":");
```

Comment:
I had not heard of `strsep` before; https://linux.die.net/man/3/strsep cautions about portability - have you verified its availability on all platforms we support?